File: liemipeli/src/test/java/com/antero/liemipeli/domain/HarjoitusTest.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.antero.liemipeli.domain;

import com.antero.liemipeli.domain.opiskelu.Harjoitus;
import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.*;

/**
 *
 * @author Antero Oikkonen
 */
public class HarjoitusTest {

    private Harjoitus harjoitus;

    public HarjoitusTest() {
    }

    @Before
    public void setUp() {
        harjoitus = new Harjoitus();
    }

    @Test
    public void testaaVastaaOikeaVastays() {

        harjoitus.setKysymys("Mitä on 2 * 2?");
        harjoitus.setOikeaVastaus("4");
        assertEquals("Oikea vastaus", harjoitus.vastaa("4"));

    }

    @Test
    public void testaaVastaaVaaraVastaus() {
        harjoitus.setKysymys("Mitä on 2 * 2?");
        harjoitus.setOikeaVastaus("4");
        assertEquals("Väärä vastaus", harjoitus.vastaa("5"));

    }
}

File: liemipeli-kotlin/src/test/kotlin/com/antero/liemipeli/domain/HarjoitusTest.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.antero.liemipeli.domain

import com.antero.liemipeli.opiskelu.Harjoitus
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

/**
 *
 * @author Antero Oikkonen
 */
class HarjoitusTest {

    private var harjoitus: Harjoitus? = null

    @BeforeEach
    fun setUp() {
        harjoitus = Harjoitus("Kuka keksi suhteellisuusteorian", "Einstain")
    }

    @Test
    fun testaaVastaaOikeaVastays() {

        harjoitus?.kysymys = "Mitä on 2 * 2?"
        harjoitus!!.setOikeaVastaus("4")
        assertTrue(harjoitus!!.vastaa("4"))

    }

    @Test
    fun testaaVastaaVaaraVastaus() {
        harjoitus!!.kysymys = "Mitä on 2 * 2?"
        harjoitus!!.setOikeaVastaus("4")
        assertFalse(harjoitus!!.vastaa("5"))

    }
}
